### DISCUSSION TITLE
Hazelcast Datasource Adapter

### DISCUSSION THREAD

**Discussion by mbrade** [getredash/redash#6825] [Ideas]
We just introduced Redash and the business likes to be able to draw their own charts using streaming data.
Unfortunately we would need to introduce a new streaming sql service and I would like to have Hazelcast adapted to it with their Jet engine. They have an own python client and speak an own SQL like dialect called HQL.
can you point me to an example how to write a new datasource adapter?

**Comment by justinclift** [comment]
@eradman Any interest in giving some pointers here? :smile:

**Comment by eradman** [comment]
@mbrade Look at [redash/query_runner/](https://github.com/getredash/redash/tree/master/redash/query_runner) for examples. Likewise tests are in [tests/query_runner/](https://github.com/getredash/redash/tree/master/tests/query_runner)

**Comment by devops-42** [comment]
Have prepared a [PR](https://github.com/getredash/redash/pull/6854) (currently in draft mode, some tests are missing) which adds JDBC as a datasource. Tested for hazelcast, but needs most-likely some more work on generalization for others.
